vault-secrets-webhook invalid ConfigMap name for daemonsets

I'm attempting to deploy a daemonset with the vault-secrets-webhook enabled. However, when creating the ConfigMap for the Vault Agent the webhook generates an invalid name.

The vault-secrets-webhook logs errors like the following:
```
2019/06/12 06:38:55 [ERROR] admission webhook error: ConfigMap "-vault-agent-config" is invalid: metadata.name: Invalid value: "-vault-agent-config": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```
Describing the daemonset has a similar message:

```
 Type     Reason        Age              From                  Message
  ----     ------        ----             ----                  -------
  Warning  FailedCreate  10s              daemonset-controller  Error creating: Internal error occurred: failed calling admission webhook "pods.vault-secrets-webhook.admission.banzaicloud.com": an error on the server ("{\"response\":{\"uid\":\"bfef376c-8cdc-11e9-9746-02e89addb8a2\",\"allowed\":false,\"status\":{\"metadata\":{},\"status\":\"Failure\",\"message\":\"ConfigMap \\\"-vault-agent-config\\\" is invalid: metadata.name: Invalid value: \\\"-vault-agent-config\\\": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')\"}}}") has prevented the request from succeeding
```

The vault-secrets-webhook image version is `banzaicloud/vault-secrets-webhook:0.4.17-rc.2`.

The main issue appears to be in `getConfigMapForVaultAgent`.

It looks like `name` is always empty and the ConfigMap name ends up as `-vault-agent-config`.

The Daemonset manifest is similar to the one below.

```yaml
apiVersion: apps/v1
kind: DaemonSet
metadata:
  name: fluentd
  labels:
    daemon: fluentd
spec:
  selector:
    matchLabels:
      name: fluentd
  template:
    metadata:
      labels:
        name: fluentd
      annotations:
        vault.security.banzaicloud.io/vault-addr: "https://vault.my.org"
        vault.security.banzaicloud.io/vault-role: "fluentd"
    spec:
      serviceAccountName: fluentd
      terminationGracePeriodSeconds: 30
      volumes:
      - name: var-log
        hostPath:
          path: /var/log
      - name: var-lib-docker-containers
        hostPath:
          path: /var/lib/docker/containers
      containers:
      - name: fluentd
        image: "my.org/daemonsets/fluentd:stable"
        imagePullPolicy: Always
        securityContext:
          privileged: true
        volumeMounts:
        - name: var-log
          mountPath: /var/log
        - name: var-lib-docker-containers
          readOnly: true
          mountPath: /var/lib/docker/containers
        ports:
        env:
          - name: POD_NAME
            valueFrom:
              fieldRef:
                fieldPath: metadata.name

          - name: POD_NAMESPACE
            valueFrom:
              fieldRef:
                fieldPath: metadata.namespace

          - name: FLUENT_HOSTNAME
            valueFrom:
              fieldRef:
                fieldPath: spec.nodeName
```

--- a/cmd/vault-secrets-webhook/main.go
+++ b/cmd/vault-secrets-webhook/main.go
@@ -374,8 +374,12 @@ func getConfigMapForVaultAgent(obj metav1.Object, vaultConfig vaultConfig) *core
 	if name == "" {
 		ownerReferences = obj.GetOwnerReferences()
 		if len(ownerReferences) > 0 {
-			generateNameSlice := strings.Split(ownerReferences[0].Name, "-")
-			name = strings.Join(generateNameSlice[:len(generateNameSlice)-1], "-")
+			if strings.Contains(ownerReferences[0].Name, "-") {
+				generateNameSlice := strings.Split(ownerReferences[0].Name, "-")
+				name = strings.Join(generateNameSlice[:len(generateNameSlice)-1], "-")
+			} else {
+				name = ownerReferences[0].Name
+			}
 		}
 	}
 	return &corev1.ConfigMap{

